scripts/delete_lambda_versions.py

```diff
@@ -0,0 +1,51 @@
+"""
+Delete all but the latest published version of every AWS Lambda function in the
+current deployment.
+"""
+
+import logging
+
+from azul import (
+    config,
+    reject,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.lambdas import (
+    Lambdas,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+
+def main():
+    require(config.terraform_component == '',
+            'This script cannot be run with the '
+            f'{config.terraform_component!r} component selected.')
+    delete_lambda_versions()
+
+
+def delete_lambda_versions():
+    functions = Lambdas().list_lambdas(config.deployment_stage)
+    for function in functions:
+        log.info('Fetching the published versions of %r', function.name)
+        response = aws.lambda_.list_versions_by_function(FunctionName=function.name)
+        reject('NextMarker' in response,
+               f"More than {len(response['Versions'])} versions of the "
+               f'function {function.name!r} were found.')
```

Comment:
We don't use fancy interpolations in assertions like this. This code is rarely executed, if ever, so the more complicated you make it, the more likely will it fail with weird failure modes that mask the underlying cause. I believe I've requested this type of change numerous times in your contributions.

```suggestion
        reject('NextMarker' in response, 'Unexpected paged response', response)
)
```